docs/decisions/0001-shellcheck.rst

```diff
@@ -0,0 +1,79 @@
+
+ShellCheck
+==========
+
+`ShellCheck <https://shellcheck.net>`_ seems very suitable:
+
+* It is well-known, with over 31.5k stars on GitHub. Pylint, in comparison, has 4.5k.
+* Its error codes, including suggested resolution steps, are `documented on its wiki <https://www.shellcheck.net/wiki/SC1000>`_.
+* It can be installed in one line from various `common package managers <https://github.com/koalaman/shellcheck#user-content-installing>`_ including apt and homebrew.
+* It can be integrated with `common editors <https://github.com/koalaman/shellcheck#user-content-in-your-editor>`_ including Vim and VSCode.
+* As of the writing of this ADR, it `seems to be actively maintained <https://github.com/koalaman/shellcheck/commits/master>`_.
+
+Finally, as a personal anecdote: ShellCheck has been extremely helpful to this ADR's author in `rewriting the edx-platform asset build in Bash <https://github.com/openedx/edx-platform/pull/31791>`_. In addition to catching simple mistakes like typos, ShellCheck's detailed error pages have provided understanding of important shell concepts such as `globbing and word splitting <https://www.shellcheck.net/wiki/SC2086>`_.
+
+Decision
+********
+
+Open edX developers are encouraged to use ShellCheck as part of the CI suite for repositories containing shell scripts.
+
+Consequences
+************
+
+* A configurable ShellCheck workflow template will be added to this repository. This will let developers add ShellChecking to their repository's PRs with a couple clicks.
+
+  * In order to ensure deterministic builds, the template will pin ShellCheck to a specific version (v0.9.0 at time of writing). The maintainer of this repository will strive to update the template's version pin over time. Maintainers of repositories using ShellCheck will be encouraged to update their pin over time. We expect that not all maintainers will do this, leading to lag & drift between versions; we expect that this will be OK, given that Bash itself is essentially feature-complete, so any newly-introcuded ShellCheck warnings are not expected to be critical.
+
+* The new template and this ADR will be announced on the forums and Slack.
+
+* ShellCheck will continue to be used in `openedx-atlas <https://github.com/openedx/openedx-atlas>`_.
+
+* The new ShellCheck template will be used to add checking to edx-platform PRs and master. Existing violations will be resolved or granted amnesty (that is: disabled with a cautionary comment).
+
+* We will share this ADR with the Maintainership Pilot team, which has been compiling a tenative list of repository standards, so that they may consider ShellCheck as a standard check for applicable repositories.
```

Comment:
[nit] typo
```suggestion
* We will share this ADR with the Maintainership Pilot team, which has been compiling a tentative list of repository standards, so that they may consider ShellCheck as a standard check for applicable repositories.
```